### PR TITLE
V1.9.5

### LINKED ISSUE #4: Check and update gfortran version
Pmetrics needs a function to check and potentially update gfortran, independently of the OS.

### PR DESCRIPTION
**Pull request by Siel**
Goal: provide the user with the possibility of update gfortran within Pmetrics.

This should close #4

### CODE REVIEW

File: R/getGfortran.R [hunk 1]
```diff
@@ -1,4 +1,20 @@
-.getOrUpdateGfortran <- function() {
+#' Download and install Gfortran updates
+#'
+#' @title Download and install Gfortran updates
+#' @return A boolean that represents if the latest version of Gfortran for your OS is installed.
+#' @author Michael Neely
+#' @export
+
+updateGfortran <- function() {
```
Comment (Siel) [inline R/getGfortran.R]: @mnneely What do you think about this function name?

Comment (mnneely) [reply R/getGfortran.R]: It seems like the underscore is becoming the convention for naming variables, functions, etc. in R?

```suggestion
update_gfortran <- function() {
```

Comment (Siel) [reply R/getGfortran.R]: Ok 👍

Comment (Siel) [reply R/getGfortran.R]: Should it have a prefix? like PMupdate_gfortran?

Comment (mnneely) [reply R/getGfortran.R]: Here's a nice coding convention reference.  As you can tell, I developed things a long time ago. I would not suffix items as op.1 now.  I would call it op_1.
https://www.r-bloggers.com/🖊-r-coding-style-guide/

**Comment by Siel** [comment]
Now we are supporting install_gfortran in both Mac and windows, linux would be a little bit harder because the commands will depend on the package manager the user is using.

I also included the script to remove the gfortran installation from the system. but I'm not sure how to proceed. should we also uninstall choco/brew ? @mnneely

**Comment by mnneely** [comment]
The removal is to remove other gfortran installations?  I don't think we need to remove choco/brew.

**Comment by Siel** [comment]
No, I was thinking about the user that want to uninstall all the dependencies pmetrics uses to setup gfortran

**Comment by mnneely** [comment]
So install gfortran, but then remove choco/brew.  But then they won’t be able to update gfortran.

I guess we could warn them of that fact...

___________________________________________
Michael Neely, MD, MSc, FCP
Chief, Infectious Diseases
Director, Laboratory of Applied Pharmacokinetics and Bioinformatics
Children’s Hospital Los Angeles

Professor of Pediatrics and Clinical Scholar
Keck School of Medicine
University of Southern California

On May 5, 2020, at 3:50 PM, Julian Otalvaro <notifications@github.com> wrote:

﻿

No, I was thinking about the user that want to uninstall all the dependencies pmetrics uses to setup gfortran

**Pull request merged**